mkosi/distributions/debian.py

```diff
@@ -130,10 +130,13 @@ def install_packages(cls, state: MkosiState, packages: Sequence[str]) -> None:
         policyrcd.unlink()
 
         sources = state.root / "etc/apt/sources.list"
+        sources_dir = state.root / "etc/apt/sources.list.d"
         if not sources.exists() and state.root.joinpath("usr/bin/apt").exists():
             with sources.open("w") as f:
                 for repo in cls.repositories(state, local=False):
                     f.write(f"{repo}\n")
+            for repo_dir in state.config.repo_dirs:
+                shutil.copytree(repo_dir, sources_dir, dirs_exist_ok=True)
```

Comment:
We shouldn't overwrite existing files here, this should be purely additive. I'd extend copy_path() with `overwrite` and `dereference` booleans and use that here (with overwrite disabled and dereference enabled) instead of `copytree()`